test/grouping.jl

```diff
@@ -1458,4 +1456,42 @@ end
     @test_throws KeyError gd[(a=:A, b=:X)]
 end
 
+@testset "Allow returning DataFrame() or NamedTuple() to drop group" begin
+    for i in typemin(UInt8):typemax(UInt8),
+        er in (DataFrame(), view(DataFrame(ones(2,2)), 2:1, 2:1),
+               view(DataFrame(ones(2,2)), 1:2, 2:1),
+               NamedTuple(), rand(0,0), rand(5,0))
+        df = DataFrame(a = 1:8, x1 = Bool.(collect(bitstring(UInt8(i))) .- '0'))
+        res = by(sdf -> sdf.x1[1] ? DataFrame(x1=[true]) : er, df, :a)
+        @test res == DataFrame(map(sdf -> sdf.x1[1] ? DataFrame(x1=[true]) : er, groupby(df, :a)))
+        if nrow(res) == 0
+            @test res == DataFrame(a=Int[])
+        else
+            @test res == df[df.x1, :]
+        end
+
+        res = by(sdf -> sdf.x1[1] ? (x1=[true],) : er, df, :a)
+        if typemin(UInt8) < i < typemax(UInt8)
+            @test_throws ArgumentError by(sdf -> sdf.x1[1] ? (x1=true,) : er, df, :a)
```

Comment:
Can you also test that returning e.g. `(x2=[true],)` fails?

Reply:
added

Reply:
interestingly it uncovered another bug where we called `names` on `NamedTuple` in the code in corner cases.